assets/js/admin/payment-method-restrictions.js

```diff
@@ -0,0 +1,74 @@
+jQuery( function( $ ) {
+
+	/**
+	 * If the WC core validation passes (errors removed), check our own validation.
+	 */
+	$( document.body ).on( 'wc_remove_error_tip', function( e, element, removed_error_type ) {
+		var type_error = 'i18n_zero_subscription_error';
+
+		// Exit early if it's this error that has been removed.
+		if ( removed_error_type === type_error ) {
+			return;
+		}
+
+		// We're only interested in product price input.
+		if ( ! $( element ).hasClass( 'wc_input_price' ) || $( element ).hasClass( 'wc_input_subscription_initial_price' ) ) {
```

Comment:
Any chance we could use more specific selectors? I installed the [Gift Wrapper for WooCommerce](https://woocommerce.com/products/gift-wrapper-for-woocommerce/) extension and am unable to offer free gift wrapping with these changes. (i.e. gift wrapping price field uses the `wc_input_price` class.)

<img width="968" alt="Screen Shot 2021-12-07 at 3 14 47 pm" src="https://user-images.githubusercontent.com/1527164/144966881-4738ff8f-d3df-40a0-bbfd-0acd66cb2058.png">

Reply:
Thanks for the review @aprea. I updated this logic to check more specific selectors. `#_subscription_price` IDs for simple subscription products and the `.wc_input_subscription_price` class for variations.